### ISSUE TITLE
OpenFOAM running, but Butterfly can't connect

### ISSUE THREAD

**Issue by MikkiSeidenschnur** [ladybug-analysis-tools/butterfly#172]
Hello everybody, 
first time I'm posting anything here on the forum, so be gentle with me

I realize that a lot of "installation issue"-problems have been flooding the forum, but I'll go ahead and explain my problem as well:

I have installed OpenFOAM v1606, and the OF_Start is working as it should:
![of_start](https://cloud.githubusercontent.com/assets/20814326/22520870/4321cc88-e8b6-11e6-9bcc-8f71d3317a42.PNG)

After this was installed I tried running Rhino/Grasshopper in administrator mode. Then I go to run the "Outdoor_flow" in grasshopper, having Butterfly installed on my computer. When I go to run the "Blockmesh" I get the following error:
![butterfly_error](https://cloud.githubusercontent.com/assets/20814326/22521031/dbefca82-e8b6-11e6-9c43-dba53395896c.png)

Following the suggestion of the error message "Run VirtualBox ...." gives me: 
![virtual_box_normal user](https://cloud.githubusercontent.com/assets/20814326/22521104/259a733a-e8b7-11e6-9937-3101f7b90abb.PNG)

Then I wanted to check whether it was in "Powered off" mode when opening as an administrator, and it says that the virtual machine has been "Aborted":
![virtual_box_administrator](https://cloud.githubusercontent.com/assets/20814326/22521135/43cadd22-e8b7-11e6-85b2-03a1f0486afe.PNG)

And just from my own experience: Docker does not accept "white spaces" in the user folder name. I realize someone must have said something in here about it. I wish however I would have found that from the forum and not by experience ;). 

I hope that you can help me with these issues, I can feel that my fingers are twitching to get back at running indoor CFD analysis with your fantastic tool. 

Best regards
Mikki Seidenschnur, CFD user, and HVAC engineer.

**Comment by mostaphaRoudsari** [comment]
Hi @MikkiSeidenschnur, Welcome to the camp :). Let's see if we can get this fixed together. Two quick questions:
1. Are you running Rhino as an administrator?
2. Did you try to start the default machine once you see it has been aborted? If we can get the issue with the VM fixed then the ones in Grasshopper will be easy to fix.

**Comment by MikkiSeidenschnur** [comment]
1. Yes, I read another comment here on the github page. 
- Right-clicked the desktop icon 
![rhino_administrator](https://cloud.githubusercontent.com/assets/20814326/22521772/67d25a68-e8b9-11e6-8c26-816f303d529c.PNG)

2. VM Box is running when I just double click it (If I don't run as administrator). When I do right-click it, it shows "Aborted". I just clicked right click on the machine and pressed: "Start". It gave me an error, hence not starting it:
![virtual_box_admin_start](https://cloud.githubusercontent.com/assets/20814326/22521898/d96b69d0-e8b9-11e6-94ac-0e1eff1b4ff8.PNG)

So question is, where do I go from here?. I had the same issue earlier, and therefore I reinstalled the whole OpenFOAM package, even as v1606, where I tried v1612 before that. And the issue persists through both versions. 

just to get the standard issues out of the way:
- I have enabled virtualization. 
- I made sure Hyper-V is disabled from the control panel.
- I am running on Windows 10.

**Comment by saket411** [comment]
Hi @**MikkiSeidenschnur** 
I remember running into similar error on my laptop. Everything looked like it was running but BF was throwing a similar error. I got it resolved by following the last step (and its sub steps) in the FAQ [here](http://www.openfoam.com/download/install-binary-windows.php)
See if it works for you as well.

**Comment by MikkiSeidenschnur** [comment]
Hi @saket411 
I've followed the last step of this FAQ. I wrote the `docker images` and I could see that the OF layer was loaded into the environment. 

I should probably try to run OpenFOAM without Butterfly to see if it still has issues. I'll try the other steps of the FAQ later today. I will probably also try to install OpenFOAM v4.0 later today with the use of ubuntu bash for windows 10. 

By the way, @mostaphaRoudsari I made a report last month with simulation comparisons from Butterfly/OpenFOAM with Ansys Fluent for indoor environment simulations. Would you be interested in seeing the report, or should I just post it somewhere here on the forum?

I will make sure to update you if I find a solution for these issues I'm having and try to document it,
Best regards,
Mikki Siedenschnur

**Comment by DambronOlivier** [comment]
Hi @MikkiSeidenschnur,
I have encountered the same issues as you. In order to solve this , I updated BF and closed gh and rhino.
Right clicked on the VM in oracle and removed. You don't have to necessarily delete all files , removing  should be good.

Relaunch docker in admin, reload the image. Relaunch create container in admin, relaunch start container in admin.
Go into oracle power off default VM, and adjust settings for memory and CPU. relaunch start container in admin.

Let us know. Hope this solves your issue.

Kind Regards,
Olivier

**Comment by MikkiSeidenschnur** [comment]
Hi @DambronOlivier 

So I tried to do as you said, but didn't get very far. I right clicked on the VM in Oracle, and removed it - I also deleted the files with it.

Then I relaunched Docker to try and load the image again. Which game me the following error: 
![docker_error](https://cloud.githubusercontent.com/assets/20814326/22546021/7af83ffe-e93b-11e6-9dbb-1639629d4333.PNG)

So now I'm wondering whether I will have to reinstall OpenFOAM again? 

Best regards
Mikki

**Comment by DambronOlivier** [comment]
@MikkiSeidenschnur,


Perhaps it would be the opportunity to install the latest version of Openfoam.

Keep us posted.
Olivier

**Comment by MikkiSeidenschnur** [comment]
@DambronOlivier Just a question while I'm installing the new version. In the OpenFOAM [README](http://www.openfoam.com/download/txt/openfoam-windows-readme_v1612+.txt) it says that Powershell v2 or above is a prerequisite, and when I'm installing OpenFOAM, notice the top bar: 
![powershell_install](https://cloud.githubusercontent.com/assets/20814326/22546897/4893c7fa-e93f-11e6-8888-aed960e7ba8a.PNG). Could this be a problem ? 

/Mikki

**Comment by MikkiSeidenschnur** [comment]
Scratch that last comment, not an issue ;). 

I managed to get the test simulation running for the outdoor case. Turns out that, when I reinstalled the whole package, and ran the Create_env as admin and Create_OF as admin, it seems to be working. 

Not entirely sure why this fixes the issue since I already have admin rights on this computer (It's my personal computer, and I have indeed checked whether I'm in the admin group in the settings of Windows 10). 

Anyways, the issue can be closed now, since my problem has been fixed. Thank you for your help everyone.

When I go along with my simulation studies for the indoor environment and Butterfly I will keep you updated on the results.
